File: Data Structures & Algorithms/jump-game-ii/submission-1.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        if (nums.size() == 1) {
            return 0;
        }

        int pos = 0, jumps = 0;

        while (pos + nums[pos] < nums.size()-1) {
            int argmax_pos = pos;
            int max_range = pos;
            for (int i = pos+1; i <= pos + nums[pos]; i++) {
                int range = i + nums[i];
                if (max_range < range) {
                    argmax_pos = i;
                    max_range = range;
                }
            }
            jumps++;
            pos = argmax_pos;
        }

        return jumps + 1;
    }
};

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        z = nums[0]
        l, r = 0, len(nums) - 1

        min_val = z

        while l <= r:
            m = l + (r-l)//2
            if nums[m] >= z:
                l = m+1
            else:
                min_val = min(min_val, nums[m])
                r = m-1

        return min_val